Source/CurrencyCloud/Client.cs

```diff
@@ -280,31 +264,22 @@ public async Task OnBehalfOf(string id, Func<Task> function)
         /// <summary>
         /// Gets a value that indicates whether the client is initialized.
         /// </summary>
-        public bool IsInitialized
-        {
-            get
-            {
-                return httpClient != null;
-            }
-        }
+        public bool IsInitialized => httpClient != null && httpClient.DefaultRequestHeaders.Contains("X-Auth-Token");
```

Comment:
It seems that `httpClient `will always be non-null, you check in the constructor